pkg/parser/parser.go

```diff
@@ -220,6 +221,9 @@ func (p *Parser) Parse(ctx context.Context, config string) (result *Result, err
 	if err := matrix.ExpandMatrices(tree); err != nil {
 		return nil, err
 	}
+	if err := merger.MergeMapEntries(tree); err != nil {
+		return nil, err
+	}
```

Comment:
Have you investigate an option to do it while yaml to node conversion. There is already some logic about alias around this line https://github.com/cirruslabs/cirrus-cli/blob/378b674f84f6092b74c33cc22e23422c6594fdb1/pkg/parser/node/yaml.go#L126

Reply:
I've initially started with the approach you're suggesting and found out that a separate single-pass makes things simpler to reason about.

It's not enough to implement some logic in `MergeMapsOrOverwrite`, `convert()` in `yaml.go` will need to be changed too because we'll need to re-evaluate the logic in `merger.go` every time a new map element is inserted and overwrite `Children` slice every time such case is detected.